src/instructlab/utils.py

```diff
@@ -438,49 +437,24 @@ def read_taxonomy_file(file_path: str, yaml_rules: Optional[str] = None):
                     parsed_p = p.split(delim)[1]
                     lint_messages.append(parsed_p)
                 logger.error("\n".join(lint_messages))
-                return None, warnings, errors
+                return warnings, errors
 
         validation_errors = validate_yaml(contents, taxonomy_path)
         if validation_errors:
             errors += validation_errors
-            return None, warnings, errors
+            return warnings, errors
 
-        # get seed instruction data
-        tax_path = "->".join(taxonomy_path.parent.parts)
-        task_description = contents.get("task_description")
-        documents = contents.get("document")
-        if documents:
-            documents = get_documents(source=documents)
```

Comment:
where is this function used now? Should we also kill it? (This will also open a path to clean up `git_clone_checkout`.)

Reply:
I think you still need to leave the reading of the documents repo. This is an important check for `taxonomy diff` since people often get this part of the yaml wrong and yaml schema validation cannot validate the URL, commit SHA, or patterns.

Reply:
If so, then the function should probably be also renamed into `validate_documents` (and changed so that it doesn't `raise SystemExit`but a more specific exception.

Reply:
Great points. I've pushed changes to make better use of an already defined custom Exception type, as well as restoring the code to make sure we can load the knowledge document content, including renaming the function to make it more clear what we care about